src/ansys/pre_commit_hooks/tech_review.py

```diff
@@ -0,0 +1,253 @@
+        if author_maint_name not in authors_name:
+            print(f"Project author name is not {author_maint_name}")
+            non_compliant = 1
+        elif author_maint_name not in maintainers_name:
+            print(f"Project maintainer name is not {author_maint_name}")
+            non_compliant = 1
+
+        # Check the project author and maintainer email is pyansys.core@ansys.com
+        authors_email = config["project"]["authors"][0]["email"]
+        maintainers_email = config["project"]["maintainers"][0]["email"]
+        if author_maint_email not in authors_email:
+            print(f"Project author email is not {author_maint_email}")
+            non_compliant = 1
+        elif author_maint_email not in maintainers_email:
+            print(f"Project maintainer email is not {author_maint_email}")
+            non_compliant = 1
+
+    return non_compliant, name
+
+
+def check_setup_py(author_maint_name: str, author_maint_email: str, non_compliant: int):
+    """Check setup.py file for correct naming convention, version, author, and maintainer."""
+    print("Not implemented")
+
+
+def check_file_exists(files: list, project_name: str, start_year: str) -> int:
+    """Check files exist. If they do not exist, create them using jinja templates."""
+    # check if there is a readme (markdown or rst)
+    non_compliant = 0
+    year_str = (
+        start_year if start_year == DEFAULT_START_YEAR else f"{start_year} - {DEFAULT_START_YEAR}"
+    )
+
+    for file in files:
+        repo_file_path = os.path.join(REPO_PATH, file)
+        file_content = generate_file_from_jinja(file, project_name, year_str)
+
+        if not os.path.exists(repo_file_path):
+            non_compliant = 1
+            print(f"{file} does not exist. Creating file from template...")
+
+            # Create the missing file using jinja templates
+            with open(repo_file_path, "w") as f:
+                f.write(file_content)
+        else:
+            # Check content
+            if file == ("CONTRIBUTORS.md" or "LICENSE"):
+                check_file_content(repo_file_path, file_content)
+
+    return non_compliant
+
+
+def generate_file_from_jinja(file, project_name, year_str):
+    """Generate file using jinja templates."""
+    loader = FileSystemLoader(searchpath=os.path.join(HOOK_PATH, "templates"))
+    env = Environment(loader=loader)
+    template = env.get_template(file)
+    file_content = template.render(
+        doc_repo_name=DOC_REPO_NAME, project_name=project_name, year_span=year_str
+    )
+
+    return file_content
+
+
+def check_file_content(file, generated_content):
+    """Check file content."""
+    non_compliant = 0
+
+    generated_file = NamedTemporaryFile(mode="w", delete=False).name
+    with open(generated_file, "w") as f:
+        f.write(generated_content)
+
+    # Check if CONTRIBUTORS.md content has been changed from template
+    if file == "CONTRIBUTORS.md" and check_same_content(file, generated_file):
```

Comment:
Have you considered using an Enum class for the file names? Seems to me like a more scalable and robust way to handle all these conditional checks

Reply:
Sure I can try, I haven't used Enums before